kernel/src/syscall.rs

```diff
@@ -616,6 +616,150 @@ fn return_scalar5(
+                    pid,
+                    tid,
+                    result: xous_kernel::Result::MemoryReturned(MemorySize::new(arg3), MemorySize::new(arg4)),
+                }
+            }
+            WaitingMessage::MovedMemory => {
+                println!(
+                    "WARNING: Tried to wait on a scalar message that was actually moved memory"
+                );
+                return Err(xous_kernel::Error::ProcessNotFound);
+            }
+            WaitingMessage::None => {
+                println!("WARNING: Tried to wait on a message that didn't exist");
+                return Err(xous_kernel::Error::ProcessNotFound);
+            }
+        };
+
+        // If there is a pending message, fetch it and schedule the thread to run
+        if let Some(msg) = next_message {
+            if cfg!(baremetal) {
+                // When the server is resumed, it will receive this as a return value.
+                ss.set_thread_result(
+                    server_pid,
+                    server_tid,
+                    xous_kernel::Result::MessageEnvelope(msg),
+                )?;
+                ss.ready_thread(response.pid, response.tid).unwrap();
+                ss.set_thread_result(response.pid, response.tid, response.result)?;
+                ss.activate_process_thread(server_tid, response.pid, response.tid, true)
+                    .map(|_| Ok(xous_kernel::Result::ResumeProcess))
+                    .unwrap_or(Err(xous_kernel::Error::ProcessNotFound))
+            } else {
+                println!("Rescheduling server because of full queue");
+
+                // Switch to the client and return the result
+                ss.ready_thread(response.pid, response.tid)?;
+                ss.switch_to_thread(response.pid, Some(response.tid))?;
+                ss.set_thread_result(response.pid, response.tid, response.result)?;
+
+                // Return the new message envelope to the server
+                Ok(xous_kernel::Result::MessageEnvelope(msg))
+            }
+        } else {
+            // For baremetal targets, switch away from this process.
+            if cfg!(baremetal) {
+                // Switch back to the client process.
+                ss.ready_thread(response.pid, response.tid)?;
+
+                // Set the thread result for the client
+                ss.set_thread_result(response.pid, response.tid, response.result)?;
+
+                // Activate the client thread and switch to it
+                ss.activate_process_thread(server_tid, response.pid, response.tid, false)
+                    .map(|_| Ok(xous_kernel::Result::ResumeProcess))
+                    .unwrap_or(Err(xous_kernel::Error::ProcessNotFound))
+            }
+            // For hosted targets, simply return `BlockedProcess` indicating we'll make
+            // a callback to their socket at a later time.
+            else {
+                println!("Unscheduling server due to empty queue");
```

Comment:
this seems to introduce prints quite frequently. OK to comment out? or is there some other debug facility I should be using.

Reply:
Whoops, looks a print statement escaped pruning. I've removed that.